### PR TITLE
fix: config variables: bugs with serialization + types

### PR DESCRIPTION
**Pull request by razbensimon**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
have bugs with the serialization of configuration variable and its schema

### Solution
fix the serialization 
fix types
add tests
remove redundant gray code

### CODE REVIEW

**Comment by razbensimon** [comment]
[Configuration Variables serialization](https://trello.com/c/OeuxG4Q5/3438-configuration-variables-serialization)

File: client/model_test.go [hunk 1]
```diff
@@ -26,4 +25,23 @@ var _ = Describe("Models", func() {
 			Entry("No value", nil, BeNil()),
 		)
 	})
+
+	Describe("ConfigurationVariable", func() {
+		Describe("Deserialize", func() {
```
Comment (RLRabinowitz) [inline client/model_test.go:30]:
```suggestion
		Describe("Schema field", func() {
```

File: client/model.go [hunk 2]
```diff
@@ -39,20 +39,20 @@ type ProjectCreatePayload struct {
...
 	Value          string                      `json:"value"`
 	OrganizationId string                      `json:"organizationId"`
 	UserId         string                      `json:"userId"`
 	IsSensitive    bool                        `json:"isSensitive"`
-	Scope          string                      `json:"scope"`
+	Scope          Scope                       `json:"scope"`
 	Id             string                      `json:"id"`
 	Name           string                      `json:"name"`
-	Type           int64                       `json:"type"`
+	Type           ConfigurationVariableType   `json:"type"`
```
Comment (RLRabinowitz) [inline client/model.go:55]: Maybe add tests on the serialization of these?

**Review by RLRabinowitz — COMMENTED** [review]
Nice 💪🏻 
Left a couple comments

File: client/model_test.go (hunk 1 above)
Comment (RLRabinowitz) [reply client/model_test.go]: https://github.com/env0/terraform-provider-env0/pull/115/commits/fa1968507f41c838973354bca80c34da8d1d233f

File: client/model.go (hunk 2 above)
Comment (RLRabinowitz) [reply client/model.go:55]: https://github.com/env0/terraform-provider-env0/pull/115/commits/fa1968507f41c838973354bca80c34da8d1d233f

**Review by RLRabinowitz — APPROVED** [review]

**Pull request merged**